### PR TITLE
Add dirty patch check

### PR DESCRIPTION
**Pull request by lipnitsk**
As discussed with @neheb and @aparcar on #openwrt-devel and https://github.com/openwrt/packages/pull/14646

Signed-off-by: Ilya Lipnitskiy <ilya.lipnitskiy@gmail.com>

### CODE REVIEW

File: entrypoint.sh [hunk 1]
```diff
@@ -63,6 +63,22 @@ else
 			echo "Package HASH check failed"
 			exit 1
 		fi
+
+		PATCHES_DIR=`find feeds -path "*/$PKG/patches"`
```
Comment (neheb) [inline entrypoint.sh:67]: shellcheck will error on this, asking to replace with $(). I don't know if that will break it though.

File: entrypoint.sh [hunk 2]
```diff
@@ -63,6 +63,22 @@ else
...
+		if [ -d "$PATCHES_DIR" ]; then
+			make \
+				BUILD_LOG="$BUILD_LOG" \
+				IGNORE_ERRORS="$IGNORE_ERRORS" \
+				QUILT_REFRESH_ARGS="--no-timestamps --no-index -p ab" \
```
Comment (neheb) [inline entrypoint.sh:72]: Isn't this default?

Comment (lipnitsk) [reply entrypoint.sh:73]: it is listed on https://openwrt.org/docs/guide-developer/build-system/use-patches-with-buildsystem#prepare_quilt_configuration, I wasn't sure - can remove if not needed.

File: entrypoint.sh (hunk 1 above)
Comment (lipnitsk) [reply entrypoint.sh:67]: you mean the backticks? Can use $()

File: entrypoint.sh (hunk 2 above)
Comment (neheb) [reply entrypoint.sh:73]:
```
~/d/openwrt > git grep no-timestamps
include/quilt.mk:               QUILT_DIFF_OPTS="-p" $(QUILT_CMD) refresh -p ab --no-index --no-timestamps; \
```

Comment (lipnitsk) [reply entrypoint.sh]: Removed QUILT options.

**Review by neheb — APPROVED** [review]

File: entrypoint.sh [hunk 3]
```diff
@@ -40,13 +40,13 @@ else
 		make \
 			BUILD_LOG="$BUILD_LOG" \
 			IGNORE_ERRORS="$IGNORE_ERRORS" \
-			"package/$PKG/download" V=s || \
+			"package/$PKG/download" V="$V" || \
```
Comment (aparcar) [inline entrypoint.sh:43]: Sorry now I remember why this was hardcoded to `s`: You should always have the full "debug" log for the checks, but you likely don't want the the full debug log if you build a ton of packages.

Please remove this commit again.

**Review by aparcar — CHANGES_REQUESTED** [review]

File: entrypoint.sh (hunk 3 above)
Comment (lipnitsk) [reply entrypoint.sh]: okay, done

**Review by neheb — APPROVED** [review]

**Comment by aparcar** [comment]
I tried to trigger the check via the following PR but it doesn't work, any ideas?
https://github.com/openwrt/packages/pull/14783

**Comment by lipnitsk** [comment]
I think the reason is that the package feed is at /home/runner/work/packages/packages/utils/podman, whereas OpenWrt build dir is at /home/build/openwrt and the script is looking under ./feeds, let me see if I can generify it a bit..

**Pull request merged**